src/utilities/hoist.ts

```diff
@@ -0,0 +1,47 @@
+const REACT_LIFECYCLE_EXCLUSIONS = [
+  'setState',
+  'forceUpdate',
```

Comment:
Hmm there's a bit of inconsistency then because forceUpdate on withViewport has different semantics from the other decorators..